cmake/Embed.cmake

```diff
@@ -97,23 +95,23 @@ function(generate_embed_source EMBED_NAME)
         string(REGEX REPLACE ".[A-Za-z0-9_]+$" "" BASE_NAME ${BASE_NAME})
 
         string(APPEND INIT_KERNELS "
-            { \"${BASE_NAME}\", { ${START_SYMBOL}, ${END_SYMBOL}} },
+            { \"migraphx/kernels/${BASE_NAME}\", { ${START_SYMBOL}, ${LENGTH_SYMBOL} } },
         ")
     endforeach()
 
     file(WRITE "${PARSE_HEADER}" "
-#include <unordered_map>
-#include <string>
 #include <utility>
-const std::unordered_map<std::string, std::pair<const char*,const char*>>& ${EMBED_NAME}();
+#include <string_view>
+#include <vector>
+std::vector<std::pair<std::string_view, std::string_view>> ${EMBED_NAME}();
```

Comment:
Initially, I thought the same - you are using the map to speed up the search. However, when I found the place you call the function `migraphx_kernels()`, I discovered the result is just repacked from `std::map` to `std::vector`, and later `std::vector` is used all the time. So I replaced `std::map` with `std::vector` for RVO to kick in.

Reply:
> Initially, I thought the same - you are using the map to speed up the search. However, when I found the place you call the function migraphx_kernels(), I discovered the result is just repacked from std::map<pair<...>> to std::vector<src_file>, and later std::vector<src_file> is used all the time.

That is the case for the kernel files, but when we add the tuning configs then that will lookup a file by name.

> So I replaced std::map with std::vector for RVO to kick in (on Windows only - on Linux, you still need to translate from pair to src_file).

To optimize it, it would be better to do this transformation once and assign it to a local static variable. We do something similar for CK headers as well where we insert pragmas for the headers.